desc/compute/_stability.py

```diff
@@ -398,15 +402,49 @@ def _ideal_ballooning_gamma2(params, transforms, profiles, data, **kwargs):
 
     A_redo = B_inv @ A @ jnp.transpose(B_inv, axes=(0, 1, 3, 2))
 
-    w, _ = jnp.linalg.eigh(A_redo)
-    # max over "zeta" axis, still a function of rho, alpha, zeta0
-    gamma = jnp.real(jnp.max(w, axis=(2,)))
+    # TODO: Issue #1750
+    # Try jax.scipy.eigh_tridiagonal or a better solver for improved performance
+    w, v = jnp.linalg.eigh(A_redo)
+
+    flat_w = w.flatten()
+
+    # Find the top_k eigenvalues.
+    top_eigvals, top_flat_indices = jax.lax.top_k(flat_w, k=Neigvals)
 
-    data["ideal ballooning lambda"] = gamma.flatten()
+    # convert back to original shape
+    alpha_idx, zeta0_idx, theta_idx = jnp.unravel_index(top_flat_indices, w.shape)
+
+    # v becomes less than the machine precision at some theta points which gives NaNs
+    # stop_gradient prevents that. Not sure how it will affect an objective that
+    # requires both the eigenvalue and eigenfunction
```

Comment:
Would using https://docs.jax.dev/en/latest/_autosummary/jax.lax.linalg.eigh.html#jax.lax.linalg.eigh be simpler and solve this issue?

Reply:
If you try it, this is what you'll get

│NotImplementedError: subset_by_index not supported on CPU and GPU

Reply:
It's funny because there is no warning in their docs.